src/main/java/appeng/me/storage/ItemCellInventory.java

```diff
@@ -0,0 +1,295 @@
+			{
+				final IAEItemStack r = input.copy();
+				r.setStackSize( r.getStackSize() - remainingItemCount );
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + remainingItemCount );
+					this.updateItemCount( remainingItemCount );
+					this.saveChanges();
+				}
+				return r;
+			}
+			else
+			{
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + input.getStackSize() );
+					this.updateItemCount( input.getStackSize() );
+					this.saveChanges();
+				}
+				return null;
+			}
+		}
+
+		if( this.canHoldNewItem() ) // room for new type, and for at least one item!
+		{
+			final int remainingItemCount = (int) this.getRemainingItemCount() - this.getBytesPerType() * itemsPerByte;
+			if( remainingItemCount > 0 )
+			{
+				if( input.getStackSize() > remainingItemCount )
+				{
+					final IAEItemStack toReturn = input.copy();
+					toReturn.setStackSize( input.getStackSize() - remainingItemCount );
+					if( mode == Actionable.MODULATE )
+					{
+						final IAEItemStack toWrite = input.copy();
+						toWrite.setStackSize( remainingItemCount );
+
+						this.cellItems.add( toWrite );
+						this.updateItemCount( toWrite.getStackSize() );
+
+						this.saveChanges();
+					}
+					return toReturn;
+				}
+
+				if( mode == Actionable.MODULATE )
+				{
+					this.updateItemCount( input.getStackSize() );
+					this.cellItems.add( input );
+					this.saveChanges();
+				}
+
+				return null;
+			}
+		}
+
+		return input;
+	}
+
+	@Override
+	public IAEItemStack extractItems( final IAEItemStack request, final Actionable mode, final IActionSource src )
+	{
+		if( request == null )
+		{
+			return null;
+		}
+
+		final long size = Math.min( Integer.MAX_VALUE, request.getStackSize() );
+
+		IAEItemStack Results = null;
+
+		final IAEItemStack l = this.getCellItems().findPrecise( request );
+		if( l != null )
+		{
+			Results = l.copy();
+
+			if( l.getStackSize() <= size )
+			{
+				Results.setStackSize( l.getStackSize() );
+				if( mode == Actionable.MODULATE )
+				{
+					this.updateItemCount( -l.getStackSize() );
+					l.setStackSize( 0 );
+					this.saveChanges();
+				}
+			}
+			else
+			{
+				Results.setStackSize( size );
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() - size );
+					this.updateItemCount( -size );
+					this.saveChanges();
+				}
+			}
+		}
+
+		return Results;
+	}
+
+	@Override
+	public IStorageChannel getChannel()
+	{
+		return AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class );
+	}
+
+	protected void loadCellItem( NBTTagCompound compoundTag, int stackSize )
+	{
+
+		// Now load the item stack
+		final ItemStack t;
+		try
+		{
+			t = new ItemStack( compoundTag );
+			if( t.isEmpty() )
+			{
+				AELog.warn( "Removing item " + compoundTag + " from storage cell because the associated item type couldn't be found." );
+				return;
+			}
+		}
+		catch( Throwable ex )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1181)